tutorials/send-email-with-sparkpost-and-nodejs-on-google-app-engine.md

```diff
@@ -0,0 +1,157 @@
+
+## Prepare
+
+1. Sign up for a [SparkPost account](https://app.sparkpost.com/sign-up).
+
+1. Add a [new sending domain](https://app.sparkpost.com/account/sending-domains).
+
+1. [Create an API key](https://app.sparkpost.com/account/credentials) with the "Transmissions: Read/Write" privilege.
+
+1. Initialize a `package.json` file with the following command:
+
+    ```sh
+    npm init
+    ```
+
+1. Install some dependencies, including the [SparkPost Node.js client](https://github.com/sparkpost/node-sparkpost):
+
+    ```sh
+    npm install --save express body-parser pug sparkpost
+    ```
+
+## Create
+
+1. Create an `index.js` file with the following contents:
+
+    ```js
+    'use strict';
+
+    const express = require('express');
+    const path = require('path');
+    const bodyParser = require('body-parser');
+
+    const SparkPost = require('sparkpost');
+    const spClient = new SparkPost(process.env.SPARKPOST_API_KEY);
+
+    const app = express();
+
+    // Setup view engine
+    app.set('views', path.join(__dirname, 'views'));
+    app.set('view engine', 'pug');
+
+    // Parse form data
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({ extended: false }));
+
+    app.get('/', (req, res) => res.render('index'));
+
+    app.post('/hello', (req, res, next) => {
+      spClient.transmissions.send({
+        options: { sandbox: true },
+        content: {
+          from: 'appengine-node-demo@sparkpostbox.com',
+          subject: 'Hello from Google AppEngine!',
+          html:'<html><body><p>Google AppEngine + Node.js + SparkPost = awesome!</p></body></html>'
+        },
+        recipients: [
+          {address: req.body.email} 
+        ]
+      }).then(result => {
+        res.render('index', {sent: true});
+      }).catch(err => {
+        res.render('index', {err: err});
+        console.error(err);
+      });
+    });
+    
+    app.listen(process.env.PORT || 8080);
+    ```
+
+1. Create a directory named `views`:
+
+    ```sh
+    mkdir views
+    ```
+
+1. Create a file named `index.pug` inside the `views` directory with the
+following contents:
+
+    ```
+    doctype html
+    html    
+      head
+        title= title
+      body
+        h1 hello world!
+        p express.js + sparkpost on google app engine.
+        hr
+        if sent
+          p email sent!
+        if err
+          p Oh my. Something's not right:
+            ul
+              each e in err.errors
+                li
+                  strong=e.message+': '
+                  |#{e.description}
+        else
+          form(name="hello", action="/hello", method="post")
+            input(type="email", placeholder="enter your email to send yourself a hello world message", name="email", style="width: 50%; margin-right: 15px;")
+            input(type="submit", value="send")
+    ```
+
+## Run
+
+1. Run the app with the following command:
+
+    ```sh
+    SPARKPOST_API_KEY=your-sparkpost-api-key npm start
```

Comment:
Actually, I think I'll change the instructions to use `server.js` since [npm start defaults to `node server.js`](https://docs.npmjs.com/cli/start). That aligns with the sample app nicely too.